native/core/src/events/common/object_reference.rs

```diff
@@ -164,6 +207,336 @@ impl RefDetails {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct RefDetailsBuilder {
+    ref_details: RefDetails,
+}
+
+impl RefDetailsBuilder {
+    pub fn new_task_ref_builder(target_id: OwnedEventId, task_list: OwnedEventId) -> Self {
+        let ref_details = RefDetails::Task {
+            target_id,
+            room_id: None,
+            task_list,
+            action: TaskAction::default(),
+        };
+        RefDetailsBuilder { ref_details }
+    }
+
+    pub fn new_task_list_ref_builder(target_id: OwnedEventId) -> Self {
+        let ref_details = RefDetails::TaskList {
+            target_id,
+            room_id: None,
+            action: TaskListAction::default(),
+        };
+        RefDetailsBuilder { ref_details }
+    }
+
+    pub fn new_calendar_event_ref_builder(target_id: OwnedEventId) -> Self {
+        let ref_details = RefDetails::CalendarEvent {
+            target_id,
+            room_id: None,
+            action: CalendarEventAction::default(),
+        };
+        RefDetailsBuilder { ref_details }
+    }
+
+    pub fn new_link_ref_builder(title: String, uri: String) -> Self {
+        let ref_details = RefDetails::Link { title, uri };
+        RefDetailsBuilder { ref_details }
+    }
+
+    pub fn target_id(&mut self, target_id: String) -> &mut Self {
+        match self.ref_details.clone() {
+            RefDetails::Task {
+                room_id,
+                task_list,
+                action,
+                ..
+            } => {
+                if let Ok(target_id) = EventId::parse(target_id) {
+                    self.ref_details = RefDetails::Task {
+                        target_id,
+                        room_id,
+                        task_list,
+                        action,
+                    };
+                } else {
+                    error!("couldn't parse target_id for Task ref");
+                }
+            }
```

Comment:
why not ...
```suggestion
    pub fn target_id(&mut self, target_id: String) -> Result<()> {
        match self.ref_details.clone() {
            RefDetails::Task {
                room_id,
                task_list,
                action,
                ..
            } => {
                let target_id = EventId::parse(target_id)?;
                  self.ref_details = RefDetails::Task {
                      target_id,
                      room_id,
                      task_list,
                      action,
                  };
                }
                // ...
            }
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1369/commits/875ea6e67087ad35ac158be8385e7ca323c4c39f